### ISSUE TITLE
quoted-printable encoding seems broken

### ISSUE THREAD

**Issue by tenderlove** [mikel/mail#372]
When I try to parse an email with `Content-Transfer-Encoding: quoted-printable` and `Content-Type: text/plain; charset=windows-1252`, the mail body encoding is completely broken.

I've written a test case to demo that fails against mail version 2.4.4:

``` ruby

# encoding: utf-8

require 'mail'
require 'minitest/autorun'

class TestEncoding < MiniTest::Unit::TestCase
  MAIL = DATA.read

  def setup
    @mail = Mail.read_from_string MAIL
  end

  # I would expect the encoding of the body to either by my default_internal
  # encoding (UTF-8), or the encoding of the message (windows-1252). Instead
  # it's ascii-8bit (a.k.a. binary).
  def test_encoding
    assert_equal Encoding::WINDOWS_1252, @mail.body.to_s.encoding
  end

  # The '=F1' in the message body represents 'ñ' in windows-1252.  I would
  # expect to get that character out of the body, but instead I get a substitute
  # character.
  #
  # Basically the same problem exists with the =92 and =20 characters.
  def test_quoted_printable_characters
    quoted_print = [0xF1].pack('C').force_encoding('windows-1252')
    assert_match 'ñ', quoted_print.encode('UTF-8')
    assert_match 'ñ', @mail.body.to_s
  end
end


__END__
Return-Path: <aaron.patterson@gmail.com>
MIME-Version: 1.0
Date: Mon, 27 Feb 2012 14:59:07 -0800
From: Aaron Patterson <aaron.patterson@gmail.com>
Content-Type: text/plain; charset=windows-1252
Content-Transfer-Encoding: quoted-printable

---------- Forwarded message ----------
Subject: Se=F1or!

I=92ve

--=20
Aaron Patterson
http://tenderlovemaking.com/
```
